service-discovery/consul/src/main/java/io/smallrye/stork/servicediscovery/consul/ConsulMetadataKey.java

```diff
@@ -0,0 +1,22 @@
+package io.smallrye.stork.servicediscovery.consul;
+
+import io.smallrye.stork.MetadataKey;
+
+public enum ConsulMetadataKey implements MetadataKey {
+
+    META_CONSUL_SERVICE_ID("consul-service-id"),
```

Comment:
What I like about this approach is that we can discover the metadata, and even document them :-D